File: 프로그래머스/2/42578. 의상/의상.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String,Integer> dict = new HashMap<>();
        for(String[] i : clothes){
            dict.put(i[1],dict.getOrDefault(i[1],1) +1);
        }
        int answer = 1;
        for(String i: dict.keySet()){
            answer *= dict.get(i);
        }
        
        
        return answer -1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    let hashMap = new Map();


    clothes.forEach(e => {
        if(hashMap.has(e[1])) {
            hashMap.set(e[1], hashMap.get(e[1]) + 1);
        } else {
            hashMap.set(e[1], 1)    
        }
    })


    for(let [key, value] of hashMap) {

        answer *= value + 1;
    }


    answer -= 1;

    return answer;
}